iquip/apps/logger.py

```diff
@@ -0,0 +1,189 @@
+        """Overridden.
+        
+        Emits input signal to the connected function.
+        """
+        s = self.format(record)
+        self.signaller.signal.emit(s)
+
+
+class LoggerFrame(QWidget):
+    """Frame for logging.
+
+    Attributes:
+        logEdit: A textEdit which shows all logs.
+        clearButton: A button for clearing all logs.
+        levelBox: A comboBox for setting the logger's level.
+    """
+
+    def __init__(self, parent: Optional[QObject] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        # widgets
+        self.logEdit = QTextEdit(self)
+        self.logEdit.setReadOnly(True)
+        self.clearButton = QPushButton("Clear", self)
+        self.levelBox = QComboBox(self)
+        # layout
+        layout = QVBoxLayout(self)
+        layout.addWidget(self.logEdit)
+        layout.addWidget(self.clearButton)
+        layout.addWidget(self.levelBox)
+
+
+class ConfirmClearingFrame(QWidget):
+    """A confirmation frame for log clearing."""
+
+    confirmed = pyqtSignal()
+
+    def __init__(self, parent: Optional[QObject] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        # widgets
+        self.label = QLabel("Are you sure to clear?", self)
+        self.buttonBox = QDialogButtonBox(self)
+        self.buttonBox.addButton("OK", QDialogButtonBox.AcceptRole)
+        self.buttonBox.addButton("Cancel", QDialogButtonBox.RejectRole)
+        # connect signals
+        self.buttonBox.accepted.connect(self.buttonOKClicked)
+        self.buttonBox.rejected.connect(self.buttonCancelClicked)
+        # layouts
+        layout = QVBoxLayout(self)
+        self.setLayout(layout)
+        layout.addWidget(self.label)
+        layout.addWidget(self.buttonBox)
+
+    def buttonOKClicked(self):
+        """Called when the OK button is clicked."""
+        self.confirmed.emit()
+        self.close()
+
+    def buttonCancelClicked(self):
+        """Called when the Cancel button is clicked."""
+        self.close()
+
+
+class LoggerApp(qiwis.BaseApp):
+    """App for logging.
+
+    Manages a logger frame.
+
+    Attributes:
+        loggerFrame: A frame that shows the logs.
+        confirmFame: A frame that asks to whether to clear logs.
+        handler: A handler for adding logs to GUI. 
+    """
+
+    def __init__(self, name: str, parent: Optional[QObject] = None):
+        """Extended.
+
+        Args:
+            name: Name of the App
+        """
+        super().__init__(name, parent=parent)
+        self.loggerFrame = LoggerFrame()
+        # connect signals to slots
+        self.loggerFrame.clearButton.clicked.connect(self.checkToClear)
+        self.confirmFrame = ConfirmClearingFrame()
+        self.confirmFrame.confirmed.connect(self.clearLog)
+        self.handler = LoggingHandler(self.addLog)
+        # TODO(aijuh): Change the log format when it is determined.
+        fs ="%(levelname)s [%(name)s] [%(filename)s:%(lineno)d] %(message)s "
```

Comment:
The variable name looks vague, but it will be updated later, so you can ignore it now.

Reply:
I will make two handler and two format when implement option to add log to file. I will fix name at that PR.